### PR TITLE
[net_put] fix net_put api call, also adds `check_destination` to ignore remote file check

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

Fix net_put API call.
Adds check_destination to ignore remote file check

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

net_put
libssh

##### ADDITIONAL INFORMATION

note it's not advised to pass the path with netput call in libssh that would still be a bug.

observations - 
1 - libssh->scp_obj.get() fails to handle situations where the file is being fetched but not present. The connection fails abruptly. 

task - 
```paste below
    - name: Copy file from ansible controller to a network device
      ansible.netcommon.net_put:
        src: /home/../backup_nxos.cfg
        protocol: scp
        dest: nxos_conf.cfg
        
[nxos]
10.0.-.-

[nxos:vars]
ansible_network_os=cisco.nxos.nxos
ansible_user=****
ansible_ssh_pass=****
ansible_connection=ansible.netcommon.network_cli
ansible_network_cli_ssh_type=libssh
```
new attribute - 
```yaml
    - name: Copy file from ansible controller to a network device
      ansible.netcommon.net_put:
        src: backup/backup_ios.cfg
        protocol: scp
        dest: ios_conf.cfg
        check_destination: false  # if false does not check the remote destination about the file's existence, if true it checks
```
^ impacts paramiko and libssh both

### CODE REVIEW

File: plugins/action/net_put.py [hunk 1]
```diff
@@ -142,11 +142,10 @@ def _handle_existing_file(self, conn, source, dest, proto, timeout):
         filename = str(uuid.uuid4())
         tmp_source_file = os.path.join(cwd, filename)
         try:
-            conn.get_file(
+            conn.put_file(
```
Comment (rohitthakur2590) [inline plugins/action/net_put.py:145]: @KB-perByte I am trying to understand what _handle_existing_file  does ?
isn't it confirming the existence of the file we want to copy?
am I missing something here?

Comment (KB-perByte) [reply plugins/action/net_put.py:145]: Well for starters I feel there are multiple things wrong with the module.
The module never calls the put() API from the SCP object. 

The name of the method speaks about checking if the file exists or not, and hence get_file was used. Which again is a correct behavior, but incomplete in terms of operation. 
The PR is still expected to get a few more changes. This is there to check if put works or not.
and, it does. We have to redo a few things here to actually fix the problem.

Comment (KB-perByte) [reply plugins/action/net_put.py:145]: It gets more interesting, as parmiko works with or without the change.
I am trying to get more answers by reading the code in detail. I'll share by EOD

File: plugins/connection/libssh.py [hunk 2]
```diff
@@ -594,8 +593,13 @@ def fetch_file(self, in_path, out_path, proto="sftp"):
         elif proto == "scp":
             scp = self.ssh.scp()
             try:
-                scp.get(out_path, in_path)
-            except LibsshSCPException as exc:
+                # scp.get(out_path, in_path) # wrong call
```
Comment (NilashishC) [inline plugins/connection/libssh.py:596]: I'm fine if we remove this comment.

**Comment by KB-perByte** [comment]
Closing this in favor of https://github.com/ansible-collections/ansible.netcommon/pull/646
Regards.

**Pull request closed without merging**